### PR TITLE
Handle environment variables from podman-machine

### LINKED ISSUE #4138: Using podman-remote instead of pypodman, for podman-machine
/kind feature

Note: It's hard to use `podman-remote` instead of the previous `pypodman`,
due to the lack of environment variables (some of them missing entirelly).

Here is the current configuration:
```
export PODMAN_USER="root"
export PODMAN_HOST="127.0.0.1"
export PODMAN_PORT="38053"
export PODMAN_IDENTITY_FILE="/home/anders/.local/machine/machines/box/id_rsa"
export PODMAN_IGNORE_HOSTS="true"
```

Trying to map these into parameters, but there is no param for the `-i` flag.
Normally SSH is transferred over a local port, rather than using the VM's IP.

The current workaround is to use the PODMAN_VARLINK_BRIDGE,
but unfortunately that doesn't work for running interactive sessions with:

```
export PODMAN_VARLINK_BRIDGE="/usr/bin/ssh -F /dev/null -o ConnectionAttempts=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none -o LogLevel=quiet -o PasswordAuthentication=no -o ServerAliveInterval=60 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null tc@127.0.0.1 -o IdentitiesOnly=yes -i /home/anders/.local/machine/machines/box/id_rsa -p 38053 sudo varlink bridge"
```

```console
$ podman-remote run -it alpine
Error: client must use upgraded connection to attach
```

Would it be possible to make `podman-remote` compatible with the above variables ?
(And add a configuration option, that would pass the correct -i to the ssh command)

```go
	return fmt.Sprintf(
		`ssh -p %d -T %s@%s -- /usr/bin/varlink -A \'/usr/bin/podman --log-level=%s varlink \\\$VARLINK_ADDRESS\' bridge`,
		port, remoteConn.Username, remoteConn.Destination, logLevel)

```
One more complication is the hardcoded locations of the commands on the remote end.
For instance, when running tiny core linux then additional software is under `/usr/local`.

`sh: /usr/bin/varlink: not found`

``` console
tc@box:~$ which varlink
/usr/local/bin/varlink
tc@box:~$ which podman
/usr/local/bin/podman
```

Couldn't we just let the remote system find them in the PATH ? Workaround was runtime symlinks.

----

Eventually, once all the various above issues are worked around - it does work OK in the end...

``` console
$ podman-remote --username=$PODMAN_USER --remote-host=$PODMAN_HOST --port $PODMAN_PORT run -it alpine
Trying to pull docker.io/library/alpine...
Getting image source signatures
Copying blob sha256:9d48c3bd43c520dc2784e868a780e976b207cbf493eaff8c6596eb871cbd9609
Copying config sha256:961769676411f082461f9ef46626dd7a2d1e2b2a38e6a44364bcbecf51e66dd4
Writing manifest to image destination
Storing signatures
/ #
```

I copied my ssh key to the /root/.ssh/authorized_keys, due to the lack of -i parameter.
It also warned me that the server at _random-port_ had changed its host key: (Again)

```
The authenticity of host '[127.0.0.1]:38053 ([127.0.0.1]:38053)' can't be established.
ECDSA key fingerprint is SHA256:0697eMVSJIEHK2LXDkE9R6R7QnHxbBkdZOaavBqz0Bs.
Are you sure you want to continue connecting (yes/no)? yes
Warning: Permanently added '[127.0.0.1]:38053' (ECDSA) to the list of known hosts.
```

That is where the PODMAN_IGNORE_HOSTS comes in handy, until host keys are fixed.
It just causes the connection to use the regular workarounds for ignoring any host keys:

`-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null`

### PR DESCRIPTION
**Pull request by afbjorklund**
* Allow changing IdentityFile and to IgnoreHosts
* Avoid hard-coding path to varlink and podman
* Allow setting default parameters with env vars
Closes #4138

### CODE REVIEW

**Comment by rhatdan** [comment]
/approve
LGTM

**Comment by mheon** [comment]
/ok-to-test

LGTM on my end but I'd like @baude to have a look before we merge

File: pkg/adapter/client_unix.go [hunk 1]
```diff
@@ -14,7 +14,14 @@ func formatDefaultBridge(remoteConn *remoteclientconfig.RemoteConnection, logLev
 	if port == 0 {
 		port = 22
 	}
+	options := ""
+	if remoteConn.IdentityFile != "" {
+		options += " -i " + remoteConn.IdentityFile
+	}
+	if remoteConn.IgnoreHosts {
+		options += " -q -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
```
Comment (vrothberg) [inline pkg/adapter/client_unix.go]: Out of curiosity: Why enabling the quite mode only here?

Comment (afbjorklund) [reply pkg/adapter/client_unix.go]: Actually I didn’t have it anywhere, but got tired if the “WARNING adding host key” output on every single command

File: pkg/adapter/client_unix.go [hunk 2]
```diff
@@ -14,7 +14,14 @@ func formatDefaultBridge(remoteConn *remoteclientconfig.RemoteConnection, logLev
...
+	}
 	return fmt.Sprintf(
-		`ssh -p %d -T %s@%s -- /usr/bin/varlink -A \'/usr/bin/podman --log-level=%s varlink \\\$VARLINK_ADDRESS\' bridge`,
-		port, remoteConn.Username, remoteConn.Destination, logLevel)
+		`ssh -p %d -T%s %s@%s -- varlink -A \'podman --log-level=%s varlink \\\$VARLINK_ADDRESS\' bridge`,
```
Comment (jwhonce) [inline pkg/adapter/client_unix.go]: @afbjorklund Is there a reason for pulling the fixed paths?

Comment (afbjorklund) [reply pkg/adapter/client_unix.go]: Pulling? I removed the hard-coded paths, that missed /usr/local/bin

**Pull request merged**